.github/workflows/build-tools.yml

```diff
@@ -10,10 +10,10 @@ on:
 jobs:
   publish-image-to-dockerhub:
     name: publish to DockerHub
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
```

Comment:
Thanks for your contribution :)
do we also need to update ubuntu version? or we can just still use 18.04.
@YTGhost

Reply:
@gy95 The ubuntu-18.04 environment is also deprecated, we should switch to ubuntu-20.04 or ubuntu-22.04 instead. see here https://github.com/actions/virtual-environments/issues/6002

Reply:
ok, waiting for maintainers to approve running CI :)

Reply:
@gy95 PTAL

Reply:
there're still two left warnings: https://github.com/kubeedge/kubeedge/actions/runs/3552996307
```
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: azure/setup-helm@v1

The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```
need fixing

Reply:
fixed

Reply:
LGTM
reference doc: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/


> @gy95 The ubuntu-18.04 environment is also deprecated, we should switch to ubuntu-20.04 or ubuntu-22.04 instead. see here [actions/runner-images#6002](https://github.com/actions/runner-images/issues/6002)

@fisherxu  PTAL, and another decison, which ubuntu version should we use, ubuntu-20.04 or ubunt-22.04?